### ISSUE TITLE
Org-gcal capture templates and duplicate events with org-gcal-sync hooks

### ISSUE THREAD

**Issue by nonreligious** [kidd/org-gcal.el#99]
Was asked to create an issue based on a question from [this](https://www.reddit.com/r/emacs/comments/hdtluc/setting_up_orggcal_capture_template/) reddit thread - apologies to @telotortium who was very helpful in the thread.
 
I've tried to use your template format:  

    (setq org-capture-templates 
        `(("a" "Appointment" entry (file ,(concat org-directory "/gcal.org")) 
        "* %?\n:PROPERTIES:\n:calendar-id:\tFirst.Last@gmail.com
        \n:END:\n:org-gcal:\n%^T--%^T\n:END:\n\n" :jump-to-captured t))) 

(added whitespace to fit on screen)

and it doesn't fix my problem of duplicate events, and I'm still getting `deferred error : (wrong-type-argument stringp nil)` after capture. 

I am a bit confused about the `:jump-to-captured t` option. With your above template, the following happens after I press `Ctrl-c c` and then `a` for Appointment:

* I am prompted for the start date and time of my event, and then press `RET`

* I am prompted for the end date and time of my event (this is fine, even useful for variable event lengths) and then press `RET`

* I am prompted for the name of my event, which I type out.

* I hit `Ctrl-c Ctrl-c` to capture the event

* I am taken to `gcal.org`, where the event info is recorded with the name as the heading. At the same time, the error `deferred error : (wrong-type-argument stringp nil)` appears in the minibuffer.

As I understand it, you are suggesting that I run `org-gcal-post-at-point` on this heading. I do this, and I get the message about the evvent being posted. I save `gcal.org`.

Now I do `M-x org-agenda` and see my weekly agenda. As I do this, two things happen simultaneously:

* The new event shows up in my agenda

* I can see that `gcal.org` is now repopulated with a duplicate of my event, with the same time info but different `ETag` and `ID` in the `:Properties:` drawer. 

I refresh my calendar, and I now have two copies of the event in my agenda.

So this is (minus the time entry setup in the capture template) essentially the same scenario I was in before. What seems to be happening is that the event is synced from the org-file to the calendar, and then running `org-agenda` pulls the same event from the calendar, where it has a different `ID` stamp, and this altered copy stored in `gcal.org` is treated as a second event.

Is it possible that the issue is in how my `org-gcal` setting is configured earlier in my init file? This is what I have below, maybe the hooks are now unnecessary?:

    (use-package org-gcal
    :ensure t
    :config
    (setq org-gcal-client-id "random_stuff.apps.googleusercontent.com"
    org-gcal-client-secret "random_stuff"
    org-gcal-file-alist '(("First.Last@gmail.com" .  "~/Org/gcal.org"))))

    (add-hook 'org-agenda-mode-hook (lambda () (org-gcal-sync) ))
    (add-hook 'org-save-all-org-buffers (lambda () (org-gcal-sync) ))

**Comment by nonreligious** [comment]
Issues #62 and #68 seem related to this.

**Comment by nonreligious** [comment]
I believe the issue is more or less fixed after #106 was fixed - thanks.

The duplication still occurs when I have `org-agenda` open in one `emacs` instance, and try to do capture to `gcal.org` in another window (which is open simultaneously) -- the error seems to arise from reading from a file that has been changed on disk.
 
However, if everything is done in the same `emacs` session -- without multiple versions of the same `gcal.org` file open at the same time -- there appears to be no problem.
